examples/end-to-end-examples/sec_10k_qa/README.md

```diff
@@ -0,0 +1,190 @@
+
+The goal of this tutorial is to show how we can build a prototype of this application using the MosaicML platform from start to finish. We will cover:
+- Setting up your development environment and the MosaicML platform ([setup](#setup))
+- Downloading and processing the data for domain specific pretraining of [MPT-7b](https://huggingface.co/mosaicml/mpt-7b) ([process and upload](#1-acquiring-the-sec-10-k-data), [MDS conversion](#2-mds-conversion))
+- Finetuning MPT-7b on financial text ([finetuning](#3-finance-finetune-mpt-7b))
+- Instruction finetuning MPT-7b on the data we used to build MPT-7b-instruct ([instruct finetuning](#4-instruct-finetune-mpt-7b))
+- Deploying your finetuned MPT-7b and an Instructor embedding model for inference using the MosaicML inference service ([convert to Hugging Face](#5-convert-the-composer-checkpoint-to-a-huggingface-checkpoint), [deploy](#6-deploy-your-model-and-an-embedding-model))
+- Using the MosaicML inference integration in LangChain to connect to your deployed models ([application](#7-application-with-gradio))
+- Building a simple frontend with [gradio](https://gradio.app/) to tie everything together ([application](#7-application-with-gradio))
+- Next steps you might take ([what's next](#what-next))
+
+At the end of this tutorial, you will have a simple web application (pictured below) that answers questions about SEC Form 10-K documents using a model that you finetuned and deployed using MosaicML! All the normal caveats about language models and making things up apply, this is a simple prototype and not production ready, and there are a variety of avenues you can pursue after going through this tutorial to improve the quality and apply the MosaicML platform to your task :)
+
+![demo](web_app_screenshot.png)
+
+
+## Which MosaicML tools will we use?
+- [LLM-foundry](https://github.com/mosaicml/llm-foundry): An open-source PyTorch library of tools for training, finetuning, evaluating, and deploying LLMs for inference.
+- [Composer](https://github.com/mosaicml/composer): An open-source PyTorch library for easy large scale deep learning.
+- [Streaming](https://github.com/mosaicml/streaming): An open-source PyTorch library for efficiently and accurately streaming data from the cloud to your training job, whereever that job is running.
+- [MCLI](https://docs.mosaicml.com/projects/mcli/en/latest/): The command line interface for running training and inference jobs on the MosaicML platform.
+
+
+## Setup
+
+All commands in this tutorial are going to be run using MCLI yamls. For understanding how this works, and what is actually running, there are a few important details to understand.
+
+1) The MosaicML platform does not have permanent storage on the compute nodes. This means that all data will be streamed in/out to/from a cloud object store. See [MosaicML setup](#mosaicml-platform-setup) for more details on setting up your cloud provider of choice.
```

Comment:
We say that data will be streamed in/out but we don't say how. This might be a good place to plug Streaming Datasets so it's clear that's the component we use to accomplish this.

Reply:
added